### DISCUSSION TITLE
Quick Repeat of Failed Tests

### DISCUSSION THREAD

**Discussion by Chiramisudo** [monkeytypegame/monkeytype#5870] [Ideas]
In an effort to increase my speed and accuracy I'm using a `min accuracy` and `weakspot` from the `funbox` and would like to be able to quickly repeat the previously failed test, or optionally to toggle the `quick restart` to instead `quick repeat` a failed test.

If this would not be a priority for the maintainers, then would someone at least be willing to look at my PR if I added this feature? In that case, I've identified the following relevant source files, but given my unfamiliarity with the codebase and inexperience in Node and TypeScript, I may have missed some, but I'm willing to learn. 😀

```
frontend/src/html/pages/settings.html
frontend/src/ts/config.ts
frontend/src/ts/controllers/input-controller.ts
frontend/src/ts/pages/settings.ts
packages/contracts/src/schemas/configs.ts
```

May be related to #2627, but I'm requesting a slightly different implementation and willing to help.

**Comment by m-dwork** [comment]
I am looking for a repeat keyboard shortcut also. Looking to practice the same test a few times without having to use the mouse.

**Reply by m-dwork** [reply]
Would be nice just to have single key shortcuts on the test result screen. Keys don't do anything. Instead, just have n for next, r for repeat, for example.

**Comment by JayceWright** [comment]
I use tampermonkey for that works perfect, you need just to hit R on results screen

`// ==UserScript==
// @name         Monkeytype Repeat Test Hotkey (Fixed)
// @namespace    tampermonkey.net
// @version      1.1
// @description  Нажатие клавиши "R" для повтора теста (тот же набор слов)
// @author       You
// @match        *://monkeytype.com/*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    document.addEventListener('keydown', (e) => {
        // Проверяем, что нажата клавиша R (без Shift/Ctrl)
        // И проверяем, что фокус не в поле ввода
        if (e.code === 'KeyR' && !e.shiftKey && !e.ctrlKey && !e.altKey) {
            
            // Проверяем, не пишет ли пользователь в этот момент
            const activeElem = document.activeElement;
            const isTyping = activeElem.tagName === 'INPUT' || 
                             activeElem.tagName === 'TEXTAREA' || 
                             activeElem.getAttribute('contenteditable') === 'true';

            if (isTyping) return;

            // Ищем кнопку по ID
            const repeatBtn = document.getElementById('restartTestButtonWithSameWordset');

            if (repeatBtn) {
                repeatBtn.click();
                // Опционально: предотвращаем стандартное поведение браузера
                e.preventDefault();
            }
        }
    });
})();`

**Comment by eeriemyxi** [comment]
I use this:

```js
// ==UserScript==
// @name        Restart Key
// @namespace   Violentmonkey Scripts
// @icon        data:image/svg+xml;base64,CiAgICA8c3ZnIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmlld0JveD0iMCAwIDY0IDY0Ij4KICAgICAgPHN0eWxlPgogICAgICAgICNiZ3tmaWxsOiMyODI4Mjg7fQogICAgICAgIHBhdGh7ZmlsbDojZDc5OTIxO30KICAgICAgPC9zdHlsZT4KICAgICAgPGc+CiAgICAgICAgPHBhdGggaWQ9ImJnIiBkPSJNMCAxNlEwIDAgMTYgMGgzMnExNiAwIDE2IDE2djMycTAgMTYtMTYgMTZIMTZRMCA2NCAwIDQ4Ii8+CiAgICAgICAgPHBhdGggZD0iTTkuMDkgMjQuMXYyMS4yaDUuMTJWMzMuMXEuMjU2LTQuNjEgNC40OC00LjYxIDMuNDYuMzg0IDMuNDYgMy44NHYxMi45aDUuMTJ2LTExLjVxLS4xMjgtNS4yNSA0LjQ4LTUuMjUgMy40Ni4zODQgMy40NiAzLjg0djEyLjloNS4xMnYtMTIuMnEwLTkuNDctNy4wNC05LjQ3LTQuMjIgMC03LjA0IDMuNDYtMi4xOC0zLjQ2LTYuMDItMy40Ni0zLjQ2IDAtNi4wMiAyLjQzdi0yLjA1TTQ3IDE4Ljl2NS4xMmgtNC42MXY1LjEySDQ3djE2LjFoNS4xMnYtMTYuMWg0LjYxdi01LjEyaC00LjYxVjE4LjkiLz4KICAgICAgPC9nPgogICAgPC9zdmc+CiAgICA=
// @version     1.0.0
//
// @match       https://monkeytype.com/
// @grant       none
//
// @author      -
// @description
// ==/UserScript==

document.addEventListener('keydown', (e) => {
    if (e.code === 'KeyR' && !document.querySelector("#result").classList.contains("hidden")) {
        const repeatButton = document.querySelector('#restartTestButtonWithSameWordset');
        if (repeatButton) repeatButton.click();
    }
});
```

Hit the `r` key (no shift) when you're in the result page.
